pkg/securitycontextconstraints/sccadmission/admission.go

```diff
@@ -491,16 +496,66 @@ func shouldIgnore(a admission.Attributes) (bool, error) {
 	if pod.Spec.OS != nil && pod.Spec.OS.Name == coreapi.Windows {
 		return true, nil
 	}
-	// if this is an update, see if we are only updating the ownerRef.  Garbage collection does this
-	// and we should allow it in general, since you had the power to update and the power to delete.
-	// The worst that happens is that you delete something, but you aren't controlling the privileged object itself
-	if a.GetOperation() == admission.Update && rbacregistry.IsOnlyMutatingGCFields(a.GetObject(), a.GetOldObject(), kapihelper.Semantic) {
-		return true, nil
+
+	if a.GetOperation() == admission.Update {
+		oldPod, ok := a.GetOldObject().(*coreapi.Pod)
+		if !ok {
+			return false, admission.NewForbidden(a, fmt.Errorf("object was marked as kind pod but was unable to be converted: %v", a.GetOldObject()))
+		}
+
+		// never ignore any spec changes
+		if !kapihelper.Semantic.DeepEqual(pod.Spec, oldPod.Spec) {
+			return false, nil
+		}
+
+		// see if we are only doing meta changes that should be ignored during admission
+		// for example, the OVN controller adds informative networking annotations that shouldn't cause the pod to go through admission again
+		if shouldIgnoreMetaChanges(pod, oldPod) {
+			return true, nil
+		}
 	}
 
 	return false, nil
 }
 
+func shouldIgnoreMetaChanges(newPod, oldPod *coreapi.Pod) bool {
+	// check if we're adding or changing only annotations from the ignore list
+	for key, newVal := range newPod.ObjectMeta.Annotations {
+		if oldVal, ok := oldPod.ObjectMeta.Annotations[key]; ok && newVal == oldVal {
+			continue
+		}
+
+		if !ignoredAnnotations.Has(key) {
+			return false
+		}
+	}
+
+	// check if we're removing only annotations from the ignore list
+	for key := range oldPod.ObjectMeta.Annotations {
+		if _, ok := newPod.ObjectMeta.Annotations[key]; ok {
+			continue
+		}
+
+		if !ignoredAnnotations.Has(key) {
+			return false
+		}
+	}
+
+	// do not consider annotations for further comparisons, they were checked above
+	origNewAnnotations := newPod.ObjectMeta.Annotations
+	newPod.ObjectMeta.Annotations = oldPod.ObjectMeta.Annotations
```

Comment:
Don't mutate inputs.  This got us into trouble in validation a couple years back.  This isn't needed to make the mutatingGCFields check work.

Reply:
As discussed with @stlaz, opted for doing an object deep-copy and modifying the annotations of the copy in order to perform the controller GCFields check.